src/Temporalio/Client/Configuration/ClientConfig.cs

```diff
@@ -0,0 +1,124 @@
+        /// <param name="configSource">The data source to load from.</param>
+        /// <param name="disableFile">If true, do not load from file (only from environment).</param>
+        /// <param name="configFileStrict">If true, fail if configuration file is invalid.</param>
+        /// <param name="overrideEnvVars">Environment variables to use, or null to use system environment.</param>
+        /// <returns>Loaded configuration data.</returns>
+        public static ClientConfig Load(
+            DataSource? configSource = null,
+            bool disableFile = false,
+            bool configFileStrict = false,
+            Dictionary<string, string>? overrideEnvVars = null)
+        {
+            var runtime = Runtime.TemporalRuntime.Default.Runtime;
+            var profiles = Bridge.EnvConfig.LoadClientConfig(
+                runtime,
+                configSource ?? DataSource.FromDefault(),
+                disableFile,
+                configFileStrict,
+                overrideEnvVars);
+            return new ClientConfig(profiles);
+        }
+
+        /// <summary>
+        /// Load client connection options directly from configuration.
+        /// </summary>
+        /// <param name="profile">Name of the profile to load. If null, "default" is used.</param>
+        /// <param name="configSource">The data source to load from.</param>
+        /// <param name="disableFile">If true, do not load from file (only from environment).</param>
+        /// <param name="disableEnv">If true, disable environment variable overrides.</param>
+        /// <param name="configFileStrict">If true, fail if configuration file is invalid.</param>
+        /// <param name="overrideEnvVars">Environment variables to use, or null to use system environment.</param>
+        /// <returns>Client connection options.</returns>
+        public static TemporalClientConnectOptions LoadClientConnectOptions(
+            string? profile = null,
+            DataSource? configSource = null,
+            bool disableFile = false,
+            bool disableEnv = false,
+            bool configFileStrict = false,
+            Dictionary<string, string>? overrideEnvVars = null)
+        {
+            var profileName = profile ?? "default";
+            var clientProfile = ClientConfigProfile.Load(
+                profileName,
+                configSource,
+                disableFile,
+                disableEnv,
+                configFileStrict,
+                overrideEnvVars);
+            return clientProfile.ToConnectionOptions();
+        }
+
+        /// <summary>
+        /// Deserialize client configuration from JSON.
+        /// </summary>
+        /// <param name="json">JSON string to deserialize.</param>
+        /// <param name="options">Optional JSON serializer options.</param>
+        /// <returns>Client configuration instance.</returns>
+        public static ClientConfig FromJson(string json, JsonSerializerOptions? options = null)
```

Comment:
I don't think we need to publicly support JSON serializability of env config in any SDK

Reply:
Yes - so this was intended to be the .NET equivalent of `to_dict/from_dict` in Python. Realized later this makes more sense as `toRecord/fromRecord` than JSON